### PR TITLE
Adding Set-DesktopOptions for show/hide icons and titles

### PR DESCRIPTION
**Pull request by eeevans**
Offering the addition for setting desktop options for hiding/showing desktop icons and hiding/showing icon titles.
On new machines, in addition to showing hidden files and file extensions, I often hide desktop icons and turn on the Desktop toolbar on the taskbar.  I researched turning on the Desktop toolbar but didn't find an acceptable solution.
Thanks for consideration.

Ed

### CODE REVIEW

File: Boxstarter.WinConfig/Set-DesktopOptions.ps1 [hunk 1]
```diff
@@ -0,0 +1,46 @@
+function Set-DesktopOptions {
+<#
+.SYNOPSIS
+Sets options for the Windows Task Bar
```
Comment (gep13) [inline Boxstarter.WinConfig/Set-DesktopOptions.ps1]: @eeevans I think that this should be "Sets options for the Windows Desktop"

Comment (eeevans) [inline Boxstarter.WinConfig/Set-DesktopOptions.ps1]: Yes, I agree...copy/paste but didn't update that line

**Comment by mwrock** [comment]
This is great. Will merge as soon as I have a chance to test.

**Comment by mwrock** [comment]
I'm looking to merge this in now and push a new release.

As I'm looking at this more closely, I'd like to have the show/hide argument more explicitly indicate that it impacts icons. So I'm thinking instead of separate show/hide args, instead have another on/off   arg called `-icons`.

Does that make sense?

**Comment by eeevans** [comment]
Yes, that makes sense.  I was thinking of rounding out the other options under View so ShowDesktopIcons(On/Off), AutoArrangeIcons(On/Off), AlignIconsToGrid(On/Off), and SetIconSize (Small/Medium/Large).  But yes, it definitely needs to more explicitly indicate that it's going to impact icons.

**Comment by eeevans** [comment]
Also, should I have opened an issue to go along with this for tracking or is that not really necessary?

**Comment by mwrock** [comment]
That's not necessary. 

One thing I'm seeing is that toggling the icons on after turning them off does not work and the reg key seems to turn itself back off.

I can troubleshoot but wondering if you have seen that. Testing on 2012 r2.

**Comment by eeevans** [comment]
No, I haven't seen that but I'm on Win 7 x64 ultimate.  I looked into adding the other View items but it is a total cluster on windows.  I was trying to use procmon to track down what is going on but can't see what it's doing.

**Comment by mwrock** [comment]
No worries. I refactored the switches and will look into this tonight. Hopefully we can get it released then. Thanks!

**Pull request merged**